src/main/java/net/oauth/jsontoken/AsyncJsonTokenParser.java

```diff
@@ -101,30 +88,36 @@ public AsyncJsonTokenParser(
    * or equivalent constructor of {@link JsonToken} subclasses.
    */
   public ListenableFuture<JsonToken> verifyAndDeserialize(String tokenString) {
-    ListenableFuture<String> futureTokenString = Futures.immediateFuture(tokenString);
-    AsyncFunction<String, JsonToken> verifyFunction =
-        tokenStringInput -> {
-          JsonToken jsonToken = deserialize(tokenStringInput);
-          Function<Void, JsonToken> returnFunction = unused -> jsonToken;
-          return Futures.transform(verify(jsonToken), returnFunction, executor);
-        };
+    JsonToken token;
+    try {
+      token = deserialize(tokenString);
+    } catch (Throwable t) {
```

Comment:
Only catch Exception, we don't want to catch errors (and same in provideVerifiers).

Reply:
Fixed.